### DISCUSSION TITLE
STRING CONVERT TO DATE

### DISCUSSION THREAD

**Discussion by 5011-5011** [nextgenhealthcare/connect#5497] [Q&A] [answered]
Develop a function to convert a data that arrives into an HL7 message which arrives in a string that brings a date data. However, I do not know how to pass the value of the conversion to call it in query for data insertion.

![image](https://user-images.githubusercontent.com/98954172/198325307-58aab730-8af2-46ea-a243-4b6d1a7edbdc.png)

I wonder if there will also be a way to abbreviate it as mapping. As the example below

![image](https://user-images.githubusercontent.com/98954172/198325405-ff8aac54-8dcd-4a7a-8350-af0f68c65dc7.png)

**Comment by siddharthsharma9258** [comment]
There are many errors in your scripts. I am interested to know where did you get this syntax `+dateparts[0]` and what does it do?

I am guessing you need output date in the format of `yyyy-MM-dd` ?

**Comment by pacmano1** [comment, marked as answer]
Rather than wrtie your own, see https://github.com/nextgenhealthcare/connect-examples/tree/master/Code%20Templates/Convert%20Date%20From%20Multiple%20Formats

But I am not clear on what you want to do.   Are you wanting to create a date object from a string?

**Reply by 5011-5011** [reply]
Yes! That's what I want
